### ISSUE TITLE
UnableToLoadModule error for MacOS

### ISSUE THREAD

**Issue by JennyferChang** [JuliaIO/ImageMagick.jl#122]
Hello!

I've been running into the following error when trying to run my codebase. 

UnableToLoadModule `/Users/usr/.julia/v0.6/Homebrew/deps/usr/opt/imagemagick@6/lib/ImageMagick/modules-Q16/coders/png.la': file not found @ error/module.c/OpenModule/1302

The test summary for ImageMagick indicate errors for the following
1. Gray png
2. Color
3. Writing to a stream (PR #22)
4. show(MIME) 

I tried removing ImageMagick and readding it/rebuilding it along with doing the same with its dependencies but the error still persists. I am also pretty sure only one version of ImageMagick exists in my system. 

Any help or suggestions on how to resolve this issue would be sincerely appreciated! Thank you!

**Comment by SimonDanisch** [comment]
Hi Jennyfer,
could you try out this branch:

https://github.com/JuliaIO/ImageMagick.jl/pull/121

E.g. by doing:
```Julia
Pkg.checkout("ImageMagick")
Pkg.checkout("ImageMagick", "sd/binaryprovider")
Pkg.build("ImageMagick")
Pkg.test("ImageMagick")
```
That might fix it for you and if not help me figure out what's still wrong with the new build...

**Comment by JennyferChang** [comment]
Hi Simon,

Thanks so much for your help! Sure! Please see below. Looks like I have a few warning errors from Homebrew and the major error from ImageMagick seems to be that it cannot generate deps.jl?

I have the 0.3.3 version of BinaryProvider and 0.5.1+ version of ImageMagick. Any thoughts? 

```julia
Pkg.checkout("ImageMagick") 
INFO: Checking out ImageMagick master... 
INFO: Pulling ImageMagick latest master...
INFO: No packages to install, update or remove
```

```julia
Pkg.checkout("ImageMagick", "sd/binaryprovider")
INFO: Checking out ImageMagick sd/binaryprovider...
INFO: Pulling ImageMagick latest sd/binaryprovider...
INFO: No packages to install, update or remove
```
```julia
Pkg.build("ImageMagick")
INFO: Building Homebrew
Updated 1 tap (homebrew/core).
==> New Formulae
mkcert
==> Updated Formulae
imagemagick@6 ✔            ethereum                   osquery
adr-tools                  exploitdb                  pgloader
apache-opennlp             firebase-cli               pgroonga
arp-scan                   gegl                       rabbitmq
assh                       get_iplayer                readline
aws-sdk-cpp                gmt                        rocksdb
azure-cli                  goreleaser                 roll
baresip                    gron                       scour
bench                      hadolint                   swiftformat
camlp5                     heroku                     sysbench
cimg                       hevea                      tmuxinator-completion
clojurescript              imagemagick                vim
consul-backinator          jhipster                   webpack
digdag                     kerl                       wtf
dmd                        nats-streaming-server      youtube-dl
dub                        node
erlang                     opencv
==> Renamed Formulae
saltstack -> salt
Uninstalling fontconfig... (493 files, 3.2MB)
==> Installing fontconfig from staticfloat/juliatranslated
==> Downloading https://homebrew.bintray.com/bottles/fontconfig-2.12.6.sierra.bo
Already downloaded: /Users/jenniferchang/Library/Caches/Homebrew.jl/fontconfig-2.12.6.sierra.bottle.tar.gz
==> Pouring fontconfig-2.12.6.sierra.bottle.tar.gz
==> Regenerating font cache, this may take a while
==> /Users/jenniferchang/.julia/v0.6/Homebrew/deps/usr/Cellar/fontconfig/2.12.6/
🍺  /Users/jenniferchang/.julia/v0.6/Homebrew/deps/usr/Cellar/fontconfig/2.12.6: 493 files, 3.2MB
Warning: Already linked: /Users/jenniferchang/.julia/v0.6/Homebrew/deps/usr/Cellar/fontconfig/2.12.6
Uninstalling imagemagick@6... (1,464 files, 22.3MB)
Warning: Calling <<-EOS.undent is deprecated!
Use <<~EOS instead.
/Users/jenniferchang/.julia/v0.6/Homebrew/deps/usr/Library/Taps/staticfloat/homebrew-juliadeps/imagemagick@6.rb:139:in `caveats'
Please report this to the staticfloat/juliadeps tap!

Warning: Calling <<-EOS.undent is deprecated!
Use <<~EOS instead.
/Users/jenniferchang/.julia/v0.6/Homebrew/deps/usr/Library/Taps/staticfloat/homebrew-juliatranslated/imagemagick@6.rb:139:in `caveats'
Please report this to the staticfloat/juliatranslated tap!

==> Installing imagemagick@6 from staticfloat/juliatranslated
==> Downloading https://juliabottles.s3.amazonaws.com/imagemagick@6-6.9.7-3.sier
Already downloaded: /Users/jenniferchang/Library/Caches/Homebrew.jl/imagemagick@6-6.9.7-3.sierra.bottle.tar.gz
==> Pouring imagemagick@6-6.9.7-3.sierra.bottle.tar.gz
Warning: Calling <<-EOS.undent is deprecated!
Use <<~EOS instead.
/Users/jenniferchang/.julia/v0.6/Homebrew/deps/usr/Library/Taps/staticfloat/homebrew-juliatranslated/imagemagick@6.rb:139:in `caveats'
Please report this to the staticfloat/juliatranslated tap!

🍺  /Users/jenniferchang/.julia/v0.6/Homebrew/deps/usr/Cellar/imagemagick@6/6.9.7-3: 1,464 files, 22.3MB
Warning: Already linked: /Users/jenniferchang/.julia/v0.6/Homebrew/deps/usr/Cellar/imagemagick@6/6.9.7-3
INFO: Building FFTW
INFO: Building ImageMagick
Info: Downloading https://github.com/bicycle1885/ZlibBuilder/releases/download/v1.0.1/Zlib.v1.2.11.x86_64-apple-darwin14.tar.gz to /Users/jenniferchang/.julia/v0.6/ImageMagick/deps/usr/downloads/Zlib.v1.2.11.x86_64-apple-darwin14.tar.gz...
[16:24:08] ######################################################################## 100.0%
Info: Downloading https://github.com/SimonDanisch/LibpngBuilder/releases/download/v1.6.31/libpng.v1.0.0.x86_64-apple-darwin14.tar.gz to /Users/jenniferchang/.julia/v0.6/ImageMagick/deps/usr/downloads/libpng.v1.0.0.x86_64-apple-darwin14.tar.gz...
[16:24:11] ######################################################################## 100.0%
Info: Downloading https://github.com/SimonDanisch/LibJPEGBuilder/releases/download/v9b/libjpeg.v9.0.0-b.x86_64-apple-darwin14.tar.gz to /Users/jenniferchang/.julia/v0.6/ImageMagick/deps/usr/downloads/libjpeg.v9.0.0-b.x86_64-apple-darwin14.tar.gz...
[16:24:12] ######################################################################## 100.0%
Info: Downloading https://github.com/SimonDanisch/LibTIFFBuilder/releases/download/v4.0.9/libtiff.v4.0.9.x86_64-apple-darwin14.tar.gz to /Users/jenniferchang/.julia/v0.6/ImageMagick/deps/usr/downloads/libtiff.v4.0.9.x86_64-apple-darwin14.tar.gz...
[16:24:14] ######################################################################## 100.0%
Info: Downloading https://github.com/JuliaIO/ImageMagickBuilder/releases/download/v6.9.10-4/ImageMagick.v6.9.10-4.x86_64-apple-darwin14.tar.gz to /Users/jenniferchang/.julia/v0.6/ImageMagick/deps/usr/downloads/ImageMagick.v6.9.10-4.x86_64-apple-darwin14.tar.gz...
[16:24:17] ######################################################################## 100.0%
============================================================================[ ERROR: ImageMagick ]============================================================================

LoadError: BinaryProvider.LibraryProduct(nothing, String["libMagickWand"], :libwand, "Prefix(/Users/jenniferchang/.julia/v0.6/ImageMagick/deps/usr)") is not satisfied, cannot generate deps.jl!
while loading /Users/jenniferchang/.julia/v0.6/ImageMagick/deps/build.jl, in expression starting on line 61

==============================================================================================================================================================================

===============================================================================[ BUILD ERRORS ]===============================================================================

WARNING: ImageMagick had build errors.

 - packages with build errors remain installed in /Users/jenniferchang/.julia/v0.6
 - build the package(s) and all dependencies with `Pkg.build("ImageMagick")`
 - build a single package by running its `deps/build.jl` script

==============================================================================================================================================================================

```
```julia
Pkg.test("ImageMagick")
INFO: Computing test dependencies for ImageMagick...
INFO: Installing ZipFile v0.5.0
INFO: Testing ImageMagick

ERROR: LoadError: InitError: UndefVarError: libwand not defined
Stacktrace:
 [1] __init__() at /Users/jenniferchang/.julia/v0.6/ImageMagick/src/libmagickwand.jl:36
 [2] _include_from_serialized(::String) at ./loading.jl:157
 [3] _require_from_serialized(::Int64, ::Symbol, ::String, ::Bool) at ./loading.jl:200
 [4] _require(::Symbol) at ./loading.jl:464
 [5] require(::Symbol) at ./loading.jl:405
 [6] include_from_node1(::String) at ./loading.jl:576
 [7] include(::String) at ./sysimg.jl:14
 [8] process_options(::Base.JLOptions) at ./client.jl:305
 [9] _start() at ./client.jl:371
during initialization of module ImageMagick
while loading /Users/jenniferchang/.julia/v0.6/ImageMagick/test/runtests.jl, in expression starting on line 1
============================================================================[ ERROR: ImageMagick ]============================================================================

failed process: Process(`/Applications/Julia-0.6.app/Contents/Resources/julia/bin/julia -Cgeneric -J/Applications/Julia-0.6.app/Contents/Resources/julia/lib/julia/sys.dylib --compile=yes --depwarn=yes --check-bounds=yes --code-coverage=none --color=yes --compilecache=yes /Users/jenniferchang/.julia/v0.6/ImageMagick/test/runtests.jl`, ProcessExited(1)) [1]

==============================================================================================================================================================================
INFO: Removing ZipFile v0.5.0
ERROR: ImageMagick had test errors

**Comment by SimonDanisch** [comment]
Sorry for the delay, took a while to fix macos without having it! Should be fixed now by #121! Please try again on master!

**Comment by JennyferChang** [comment]
No worries! Been following your updates and saw. It works now. Thank you!
